### PR TITLE
[Refactor] Fixing mypy issues of `syft/`

### PR DESCRIPTION
**Pull request by khoaguin**

### CODE REVIEW

File: packages/syft/src/syft/client/domain_client.py [hunk 1]
```diff
@@ -88,28 +91,32 @@ def upload_dataset(self, dataset: CreateDataset) -> Union[SyftSuccess, SyftError
...
-            metadata.show_warnings
-            and metadata.node_side_type == NodeSideType.HIGH_SIDE.value
-        ):
-            message = (
-                "You're approving a request on "
-                f"{metadata.node_side_type} side {metadata.node_type} "
-                "which may host datasets with private information."
-            )
-            prompt_warning_message(message=message, confirm=True)
+        if self.api.connection:
```
Comment (shubham3121) [inline packages/syft/src/syft/client/domain_client.py]: We can skip checking for None in self.api.connection since self.api is casted to SyftAPI

Comment (khoaguin) [reply packages/syft/src/syft/client/domain_client.py]: But Shubham here `SyftAPI.connection` can still be None, so that's why we check it there

**Review by shubham3121 — APPROVED** [review]
Monumental work @khoaguin !!! 🔥 🥳  Much needed heavy lifting !! 🦾 
Finally we can have typing checks back 💯

File: packages/syft/src/syft/client/domain_client.py (hunk 1 above)
Comment (shubham3121) [reply packages/syft/src/syft/client/domain_client.py]: Type casted to NodeConnection: 0596219b8584c888c056f9084e0f590bc1075e7a

**Pull request merged**